### PR TITLE
Jetpack connect: Avoid connection

### PR DESCRIPTION
**Pull request by johnHackworth**
Right now we have several flows that include sending users from sites we already know that are connected through JPC, just to be able to send them to the checkout page so they can buy an upgrade.

This is clearly a hack, and not even a very good one: It's causing problems to some of the sites. When we designed JPC first, we didn't have this use-case in mind, so we didn't optimize it for this. For example, the status-check we do when we start the connection process is optimized for speed, not for being 100 accurate. We know that it's going to throw some false negatives, but until now that wasn't a big deal, it was a trade-off we made in exchange for making the whole flow quicker. 

But when this trade-off affects to users who are with their money in hand, ready to pay for an upgrade, it doesn't look so good. 

What this PR do is introduce a new parameter that can be passed to most of the entry points for JPC, that indicates that we are 100% sure that the site is already connected, and that it makes it safe to totally skip the connection bits and redirect the user directly to the plans page / checkout.  It's supposed to be used only from parts that can only be shown if the user has a connected site (for example, a JITM in wp-admin).

How to test
=========

0. You need a jetpack connected site (of course)
1. Go to https://calypso.localhost:3000/jetpack/connect/store?branch=add/jpc-skip-connection-when-selling-stuff-to-connected-sites&preSelectedSite=[site_slug] 
2. You should see the plans page... pick one
3. You should reach the checkout page right away
4. Test again 1-3, but without being logged to wp.com. The result should be the same

### CODE REVIEW

**Comment by sirreal** [comment]
Rebased since there have been a lot of build related changes recently.

File: client/jetpack-connect/auth-logged-out-form.jsx [hunk 1]
```diff
@@ -57,21 +57,23 @@ class LoggedOutForm extends Component {
...
 	renderLoginUser() {
-		const { userData, bearerToken } = this.props.jetpackConnectAuthorize;
-
-		return (
-			<WpcomLoginForm
-				log={ userData.username }
-				authorization={ 'Bearer ' + bearerToken }
-				redirectTo={ this.getRedirectAfterLoginUrl() } />
-		);
+		if ( this.props.jetpackConnectAuthorize ) {
```
Comment (sirreal) [inline client/jetpack-connect/auth-logged-out-form.jsx]: This seems unrelated to the issue being addressed by this PR, is it possible that this could be falsy?

Comment (johnHackworth) [inline client/jetpack-connect/auth-logged-out-form.jsx]: yup, because with this pr, you can get to this point without receiving a jetpackConnectAuthorize at all, redirected directly here by a link in wp-admin :)
